src/ameba/source/corrector.cr

```diff
@@ -140,58 +145,6 @@ class Ameba::Source
       @line_sizes[0...line - 1].sum + (column - 1)
     end
 
-    # Replaces the code of the given node with *content*.
-    def replace(node : Crystal::ASTNode, content)
-      replace(location(node), end_location(node), content)
-    end
-
-    # Inserts the given strings before and after the given node.
-    def wrap(node : Crystal::ASTNode, insert_before, insert_after)
-      wrap(location(node), end_location(node), insert_before, insert_after)
-    end
-
-    # Shortcut for `replace(node, "")`
-    def remove(node : Crystal::ASTNode)
-      remove(location(node), end_location(node))
-    end
-
-    # Shortcut for `wrap(node, content, nil)`
-    def insert_before(node : Crystal::ASTNode, content)
-      insert_before(location(node), content)
-    end
-
-    # Shortcut for `wrap(node, nil, content)`
-    def insert_after(node : Crystal::ASTNode, content)
-      insert_after(end_location(node), content)
-    end
-
-    # Removes *size* characters prior to the given node.
-    def remove_preceding(node : Crystal::ASTNode, size)
-      remove_preceding(location(node), end_location(node), size)
-    end
-
-    # Removes *size* characters from the beginning of the given node.
-    # If *size* is greater than the size of the node, the removed region can
-    # overrun the end of the node.
-    def remove_leading(node : Crystal::ASTNode, size)
-      remove_leading(location(node), end_location(node), size)
-    end
-
-    # Removes *size* characters from the end of the given node.
-    # If *size* is greater than the size of the node, the removed region can
-    # overrun the beginning of the node.
-    def remove_trailing(node : Crystal::ASTNode, size)
-      remove_trailing(location(node), end_location(node), size)
-    end
-
-    private def location(node : Crystal::ASTNode)
-      node.location || raise "Missing location"
-    end
-
-    private def end_location(node : Crystal::ASTNode)
-      node.end_location || raise "Missing end location"
-    end
-
```

Comment:
Why? Removing these will increase the amount of boilerplate code.

Reply:
Yeah, I was thinking about this the other day. The optimal solution though would be IMO to let `add_issue` figure it out, instead fiddling with it at the call-site.